warehouse/ingest-core/src/test/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelperTest.java

```diff
@@ -0,0 +1,364 @@
+        // and the source as well as the value field are not.
+        Assertions.assertTrue(wHelper.isWhindexField("xDestField"));
+        Assertions.assertFalse(wHelper.isWhindexField("xSourceField"));
+        Assertions.assertFalse(wHelper.isWhindexField("xTestValue1"));
+    }
+
+    /**
+     * Test that isWhindexField() correctly identifies whindex fields when the rule's delete_src_field is set to "false".
+     * <p>
+     * Only the destination field should be considered a whindex field.
+     * </p>
+     */
+    @Test
+    void testWhindexFieldIdentificationWithDeleteSrcFieldFalse() {
+        // Setup configuration with delete_src_field false.
+        Configuration config = new Configuration();
+        config.set("wiki.whindex.rules.1.value_field", "yValueField");
+        config.set("wiki.whindex.rules.1.src_field", "ySourceField");
+        config.set("wiki.whindex.rules.1.delete_src_field", "false");
+        config.set("wiki.whindex.rules.1.dst_field", "yDestField");
+        config.set("wiki.whindex.rules.1.values", "yTestValue1,yTestValue2,yTestValue3,yTestValue4");
+
+        // Initialize helper.
+        Type type = new Type("wiki", null, null, null, 0, null);
+        WhindexFieldIngestHelper wHelper = new WhindexFieldIngestHelper(type);
+        wHelper.setup(config);
+
+        // Verify that only the destination field is identified as a whindex field.
+        Assertions.assertTrue(wHelper.isWhindexField("yDestField"));
+        Assertions.assertFalse(wHelper.isWhindexField("ySourceField"));
+        Assertions.assertFalse(wHelper.isWhindexField("yTestValue1"));
+    }
+
+    /**
+     * Test that isWhindexField() works correctly when delete_src_field is not specified.
+     * <p>
+     * In this case, only the destination field should be identified as a whindex field.
+     * </p>
+     */
+    @Test
+    void testWhindexFieldIdentificationWithDeleteSrcFieldNotPresent() {
+        // Setup configuration without the delete_src_field property.
+        Configuration config = new Configuration();
+        config.set("wiki.whindex.rules.namedrule.value_field", "zValueField");
+        config.set("wiki.whindex.rules.namedrule.src_field", "zSourceField");
+        config.set("wiki.whindex.rules.namedrule.dst_field", "zDestField");
+        config.set("wiki.whindex.rules.namedrule.values", "zTestValue1,zTestValue2,zTestValue3,zTestValue4");
+
+        // Initialize helper.
+        Type type = new Type("wiki", null, null, null, 0, null);
+        WhindexFieldIngestHelper wHelper = new WhindexFieldIngestHelper(type);
+        wHelper.setup(config);
+
+        // Verify that only the destination field is recognized as a whindex field.
+        Assertions.assertTrue(wHelper.isWhindexField("zDestField"));
+        Assertions.assertFalse(wHelper.isWhindexField("zSourceField"));
+        Assertions.assertFalse(wHelper.isWhindexField("zTestValue1"));
+    }
+
+    // /**
```

Comment:
All of this commented out code should be removed, or uncommented if they are tests that need to be added.